pkg/persistence/queries.go

```diff
@@ -355,6 +355,41 @@ func checkClaimKeyBan(db queryRower, identifier string) (triesRemaining int, ban
 	return triesRemaining, banDuration, nil
 }
 
+func checkHashId(db *sql.DB, identifier string) (int64, error) {
+	var count int64
+
+	row := db.QueryRow("SELECT COUNT(*) FROM encryption_keys WHERE hash_id = ? and one_time_code IS NULL", identifier)
```

Comment:
not really a big deal but I'd collapse this into a single query that selects the OTC and switch on whether it's null.

Reply:
Fixed in #57